packages/web/src/new/models/WalletModel/WalletModel.ts

```diff
@@ -52,12 +52,37 @@ export class WalletModel extends Model {
       adaptors: observable,
       pubKey: computed,
       signer: computed,
+      isMnemonicWallet: computed,
     });
 
     this.onConnect = this.onConnect.bind(this);
     this.onDisconnect = this.onDisconnect.bind(this);
   }
 
+  get pubKey(): PublicKey {
+    return new PublicKey(this.publicKey);
+  }
+
+  get signer(): Wallet {
+    return this.selectedAdaptor as Wallet;
+  }
+
+  get isMnemonicWallet(): boolean {
+    return this.selectedAdaptor?.name === MnemonicAdapterName;
+  }
+
+  get messageSigner(): MessageSignerWalletAdapter {
+    return this.selectedAdaptor as MessageSignerWalletAdapter;
+  }
+
+  confirmPassword(password: string): Promise<boolean> {
+    if (this.selectedAdaptor?.name === MnemonicAdapterName) {
+      return (this.selectedAdaptor as MnemonicAdapter).confirmPassword(password);
```

Comment:
IMHO its better to check negative scenario and make reaction for it, like validation before some action.
so it will be better to write Error in condition